src/inspector_agent.cc

```diff
@@ -338,11 +347,68 @@ AgentImpl::~AgentImpl() {
   data_written_ = nullptr;
 }
 
+void InspectorConsoleCall(const v8::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Isolate* isolate = info.GetIsolate();
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+
+  CHECK(info.Data()->IsArray());
+  v8::Local<v8::Array> args = info.Data().As<v8::Array>();
+  CHECK_EQ(args->Length(), 3);
+
+  v8::Local<v8::Value> inspector_method =
+      args->Get(context, 0).ToLocalChecked();
+  CHECK(inspector_method->IsFunction());
+  v8::Local<v8::Value> node_method =
+      args->Get(context, 1).ToLocalChecked();
+  CHECK(node_method->IsFunction());
+  v8::Local<v8::Value> config_value =
+      args->Get(context, 2).ToLocalChecked();
+  CHECK(config_value->IsObject());
+  v8::Local<v8::Object> config_object = config_value.As<v8::Object>();
+
+  std::vector<v8::Local<v8::Value>> call_args(info.Length());
+  for (int i = 0; i < info.Length(); ++i) {
+      call_args[i] = info[i];
+  }
+
+  v8::Local<v8::String> in_call_key = OneByteString(isolate, "in_call");
+  bool in_call = config_object->Has(context, in_call_key).FromMaybe(false);
+  if (!in_call) {
+    CHECK(config_object->Set(context, in_call_key,
+        v8::True(isolate)).FromJust());
+    CHECK(!inspector_method.As<v8::Function>()->Call(context, info.Holder(),
+        call_args.size(), call_args.data()).IsEmpty());
+  }
+
+  node_method.As<v8::Function>()->Call(context, info.Holder(), call_args.size(),
+      call_args.data());
```

Comment:
inspector_method doesn't throw an exception and implemented in C++. If node console function like console.assert throws an exception then it will be thrown to outer catch block like it would be without wrapConsoleCall method.